src/dbnode/client/session.go

```diff
@@ -1264,6 +1282,8 @@ func (s *session) writeAttempt(
 	// pool if ref count == 0.
 	state.Unlock()
 	state.decRef()
+	clonedAnnotation.DecRef()
+	clonedAnnotation.Finalize()
```

Comment:
We may want to manage the lifecycle of `clonedAnnotation` via `state.decRef()` above. Please see how it is done for ids which are pooled as well. I thing the lifecycle of the annotation should be the same.

Reply:
https://github.com/m3db/m3/pull/3605/commits/f0be3c594e7cb3d37e12193743ed75ef2ff7fa48